src/layout/Tabbed.js

```diff
@@ -61,14 +62,32 @@
 
     Tabbed.prototype.enter = function (domNode, element) {
         HTMLWidget.prototype.enter.apply(this, arguments);
-        this._tabContainer = element.append("div");
         this._contentContainer = element.append("div");
+        this._contentContainer.attr("id", "content_container");
     };
 
     Tabbed.prototype.update = function (domNode, element) {
         HTMLWidget.prototype.update.apply(this, arguments);
         var context = this;
 
+        // if (this._tabContainer) {this._tabContainer.remove();}
+        // this._tabContainer = element.insert("div", this.tabLocation() === "bottom" ? null : "div");
+        // this._tabContainer.attr("class", "on_" + this.tabLocation());
+
+        this._tabContainer = element.selectAll("#tab_contanier").data([this.tabLocation()], function(d) {return d;});
+        this._tabContainer
+            .enter()
+            .insert("div", function(d1) {
+                var x = d1 === "top" ? "#content_container" : null;
+                console.log(x, d1);
+                return x;
+            })
+            .attr("class", function(d2) {
+                return "on_" + d2;
+            })
+            .attr("id", "tab_contanier")
+        ;
```

Comment:
I would set the class to be "tab_w11 top" or "tab_w11 bottom" depending - that way you can still do the selectAll on selectAll(".tab_." + this._id) for the data join and have the second class for styling...